### ISSUE TITLE
Emotes are not implemented

### ISSUE THREAD

**Issue by ipad54** [pmmp/PocketMine-MP#3645]
### Issue description

if you use emotion, other players will not see it

- Expected result: if you use emotion, other players will see it
- Actual result: if you use emotion, other players will not see it

### Steps to reproduce the issue
1. Use emotion

### OS and versions

* PocketMine-MP: 3.14.0
* PHP: 7.3
* Server OS: Ubuntu
* Game version: PE/Win10

### Plugins

None

### Crashdump, backtrace or other files

None

**Comment by TwistedAsylumMC** [comment]
It is not a bug, they are not implemented. If you want emotes, you just have to redirect the EmotePacket with the flag FLAG_SERVER to all viewers then they will work.

**Comment by Govdim** [comment]
Need broadcast EmotePacket

**Comment by Nu4h-dev** [comment]
hey,
this guy add the emote for server
just look!
https://github.com/Seyz123/Emotes

**Comment by xxAROX** [comment]
Use this:
```php
	public function onData(\pocketmine\event\server\DataPacketReceiveEvent $event): void{
		$pk = $event->getPacket();
		if ($pk instanceof \pocketmine\network\mcpe\protocol\EmotePacket) {
			var_dump($event->getPlayer()->getId());
			var_dump($pk->getEmoteId());
			var_dump($pk->getFlags());
			\pocketmine\Server::getInstance()->broadcastPacket(\pocketmine\Server::getInstance()->getOnlinePlayers(), \pocketmine\network\mcpe\protocol\EmotePacket::create($event->getPlayer()->getId(), $pk->getEmoteId(), 1 << 0));
		}
	}
```

**Comment by kenygamer** [comment]
https://github.com/Seyz123/Emotes

**Comment by LolgonHDYT** [comment]
@xxAROX i tryed to make a plugin with this but it didnt work...
`<?php

declare(strict_types=1);

namespace LolgonHDYT\test;

use pocketmine\plugin\PluginBase;
use pocketmine\network\mcpe\protocol\EmotePacket;

class Main extends PluginBase{

	public function onData(\pocketmine\event\server\DataPacketReceiveEvent $event): void{
		$pk = $event->getPacket();
		if ($pk instanceof \pocketmine\network\mcpe\protocol\EmotePacket) {
			var_dump($event->getPlayer()->getId());
			var_dump($pk->getEmoteId());
			var_dump($pk->getFlags());
			\pocketmine\Server::getInstance()->broadcastPacket(\pocketmine\Server::getInstance()->getOnlinePlayers(), \pocketmine\network\mcpe\protocol\EmotePacket::create($event->getPlayer()->getId(), $pk->getEmoteId(), 1 << 0));
		}
	}
}
`

**Comment by dktapps** [comment]
This isn't a plugin support forum.

**Comment by LolgonHDYT** [comment]
Sorry...

**Comment by twk1024** [comment]
Until this implemented to pmmp, you can use this plugin instead.
https://poggit.pmmp.io/p/DoEmote

**Comment by dktapps** [comment]
Implemented by f0661999717abc3ed5cefceb4c47b732915761f3.
